src/Files/ViewModels/ItemViewModel.cs

```diff
@@ -2121,7 +2122,11 @@ private async Task<ListedItem> AddFileOrFolderAsync(string fileOrFolderPath, str
 
             var isSystem = ((FileAttributes)findData.dwFileAttributes & FileAttributes.System) == FileAttributes.System;
             var isHidden = ((FileAttributes)findData.dwFileAttributes & FileAttributes.Hidden) == FileAttributes.Hidden;
-            if (isHidden && (!UserSettingsService.PreferencesSettingsService.AreHiddenItemsVisible || (isSystem && UserSettingsService.PreferencesSettingsService.AreSystemItemsHidden)))
+            var startWithDot = findData.cFileName.StartsWith(".");
+            if ((isHidden &&
+               (!UserSettingsService.PreferencesSettingsService.AreHiddenItemsVisible ||
+               (isSystem && UserSettingsService.PreferencesSettingsService.AreSystemItemsHidden))) ||
+               (startWithDot && UserSettingsService.PreferencesSettingsService.AreItemsStartingWithDotHidden))
```

Comment:
```suggestion
               (isSystem && !UserSettingsService.PreferencesSettingsService.ShowDotFiles))) ||
               (startWithDot && !UserSettingsService.PreferencesSettingsService.ShowDotFiles))
```

Reply:
@yaichenbaum you probably didn't want to change AreSystemItemsHidden